etc/kayobe/ansible/filter_plugins/filters.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+from ansible.errors import AnsibleFilterError
+
+class FilterModule(object):
+    def filters(self):
+        return {
+            'group_hostvars_by_var':
+            self.group_hostvars_by_var,
+            'group_hostvars_by_host':
+            self.group_hostvars_by_host
+        }
+
+    def group_hostvars_by_var(self, hostvars, var, stdout=None):
+        """
+        Returns a dictionary where the keys are values for the
+        specified var in hostvars, and the values are the hosts
+        that match that value.
+        """
```

Comment:
Would benefit a lot from an example e.g:
```suggestion
        """
        Returns a dictionary where the keys are values for the
        specified var in hostvars, and the values are a list of hosts
        that match that value.
        For example, a grouping of hosts by OS release might look like:
        example_var:
          noble:
            - node1
            - node2
          jammy:
            - node3
            - node4
            - node5
        
        """
```

Reply:
Would also help to include a description of "stdout"